src/AllergenDetctorSecondary

```diff
@@ -0,0 +1,52 @@
+import components.set.Set;
+import components.set.Set1L;
+
+/**
+ * Abstract class for the AllergenDetector component.
+ * Implements the enhanced interface using only kernel methods.
+ */
+public abstract class AllergenDetectorSecondary implements AllergenDetector {
+
+    @Override
+    public boolean isSafeToEat(String ingredientList) {
+        return !this.containsAllergen(ingredientList);
+    }
+
+    @Override
+    public boolean isCrossContaminationRisk(String facilityInfo) {
+        return facilityInfo.toLowerCase().contains("processed in facility with allergens");
+    }
+
+    @Override
+    public void alertUser(String allergen, String productName) {
+        System.out.println("ALERT: " + productName + " contains " + allergen + "! Avoid consumption.");
+    }
+
+    @Override
+    public void enterIngredientString(String ingredientList) {
+        Set<String> detectedAllergens = this.getDetectedAllergens(ingredientList);
+        if (detectedAllergens.size() > 0) {
+            System.out.println("WARNING: Allergens detected in food product!");
+            System.out.println("Detected allergens: " + detectedAllergens);
+        } else {
+            System.out.println("Safe to eat! No allergens detected.");
+        }
+    }
+
+    @Override
+    public String toString() {
+        // Use getDetectedAllergens on an empty ingredient list to mimic a view
+        return "AllergenDetector";
```

Comment:
I would probably be a little more explicit here. Otherwise, this doesn't really print anything of value.